### PR TITLE
Truncate white-space and give feedback if no result for search

### PR DESCRIPTION
**Pull request by frosch95**
solves #63

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/nuget-trends?pullRequest=123) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/nuget-trends?pullRequest=123) before we can accept your contribution.<br/><br/>:x: frosch95 [sign now](https://cla.dotnetfoundation.org/dotnet/nuget-trends?pullRequest=123)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/nuget-trends?pullRequest=123) it.</sub>

File: src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts [hunk 1]
```diff
@@ -44,13 +44,28 @@ export class SearchInputComponent implements AfterViewInit {
       distinctUntilChanged(),
       filter((value: string) => value && !!value.trim()),
       tap(() => this.isSearching = true),
-      switchMap((term: string) => this.searchNuGet(term, this.packageInteractionService.searchType)),
+      switchMap((term: string) => this.searchNuGet(term.trim(), this.packageInteractionService.searchType)),
```
Comment (joaopgrassi) [inline src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts:47]: Can you also add a test to check this, please?

File: src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts [hunk 2]
```diff
@@ -44,13 +44,28 @@ export class SearchInputComponent implements AfterViewInit {
...
       catchError((_, caught) => {
         this.toastr.error('Our servers are too cool (or not) to handle your request at the moment.');
         this.isSearching = false;
         return caught;
       }),
-      tap(() => this.isSearching = false));
+      tap((value) => {
```
Comment (joaopgrassi) [inline src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts]: Add type

File: src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts [hunk 3]
```diff
@@ -44,13 +44,28 @@ export class SearchInputComponent implements AfterViewInit {
...
+        if (value && value.length === 0) {
+          switch (this.packageInteractionService.searchType) {
+            case SearchType.NuGetPackage:
+              this.toastr.info('No packages found!');
```
Comment (joaopgrassi) [inline src/NuGetTrends.Spa/src/app/shared/components/search-input/search-input.component.ts]: This whole method is rather long already.. maybe consider adding this in a sep func? Like `handleSearchResults`

**Review by joaopgrassi — CHANGES_REQUESTED** [review]
Left a few minor comments.

**Review by joaopgrassi — APPROVED** [review]

**Comment by joaopgrassi** [comment]
Thanks for the help!

**Comment by frosch95** [comment]
You are welcome!

**Pull request merged**